egress/src/main.rs

```diff
@@ -13,6 +20,60 @@ async fn main() -> Result<(), Error> {
     let manager = PostgresConnectionManager::new_from_stringlike(connect_string, NoTls)?;
     let pool = bb8::Pool::builder().build(manager).await?;
 
+    let stinfo_conn_string = std::env::var("STINFO_CONN_STRING")?;
+
+    let db_ts_list = fetch_timeseries_list_from_database(&pool.get().await.unwrap()).await?;
+    let default_table = Arc::new(RwLock::new(
+        fetch_message_priority_default(&stinfo_conn_string).await?,
+    ));
+    let exception_table = Arc::new(RwLock::new(
+        fetch_message_priority_exception(&stinfo_conn_string).await?,
+    ));
+
+    // Filter handling (needs connection to stinfosys database, as well as to lard)
+    let filter_table = Arc::new(RwLock::new(
+        create_filter_timeseries_table(db_ts_list, default_table, exception_table).unwrap(),
+    ));
+    let mut background_filter_table = filter_table.clone();
+
+    let pool_loop = pool.clone();
+    debug!("Spawning task to refresh filter table...");
+    // background task to refresh filter table every 30 mins
+    tokio::task::spawn(async move {
+        let mut interval = tokio::time::interval(tokio::time::Duration::from_secs(30 * 60));
+
+        loop {
+            interval.tick().await;
+            info!("Refreshing filter table");
+            let conn_loop = &pool_loop.get().await.unwrap();
+            async {
+                let new_db_ts_list = fetch_timeseries_list_from_database(conn_loop)
+                    .await
+                    .unwrap();
+                let new_default_table = Arc::new(RwLock::new(
+                    fetch_message_priority_default(&stinfo_conn_string)
+                        .await
+                        .unwrap(),
+                ));
+                let new_exception_table = Arc::new(RwLock::new(
+                    fetch_message_priority_exception(&stinfo_conn_string)
+                        .await
+                        .unwrap(),
+                ));
+                let new_filter_table = Arc::new(RwLock::new(
+                    create_filter_timeseries_table(
+                        new_db_ts_list,
+                        new_default_table,
+                        new_exception_table,
+                    )
+                    .unwrap(),
+                ));
+                background_filter_table = new_filter_table;
+            }
+            .await;
+        }
```

Comment:
We are not actually refreshing the filter here though? You need something like what we have in the ingestor

```rust
let mut filter = background_filter_tables.write().unwrap();
*filter = new_filter;
```

Reply:
This also...